Objects/rangeobject.c

```diff
@@ -240,6 +240,14 @@ compute_range_length(PyObject *start, PyObject *stop, PyObject *step)
     assert(PyLong_Check(stop));
     assert(PyLong_Check(step));
 
+    /* fast path for one argument case of range */
+    if (start == zero && step == one) {
+        if (_PyLong_IsPositive((const PyLongObject *)stop)) {
+            return Py_NewRef(stop);
+        }
+        return zero;
```

Comment:
You forget a Py_NewRef.

Reply:
This was intentional as `zero` is an immortal instance.